### PR TITLE
Modify to allow multiple column select

### PR DESCRIPTION
**Pull request by kanga333**

### CODE REVIEW

File: ha_my_quiver.hpp [hunk 1]
```diff
@@ -116,14 +116,24 @@ class ha_my_quiver : public handler {
...
-      field->set_notnull();
-      field->store(value, false);
+
+    auto n_columns = table->s->fields;
+    for (auto i = 0; i < n_columns; i++) {
+      auto field = static_cast<Field_longlong*>(table->field[i]);
+      auto name = field->field_name;
+      auto column = std::static_pointer_cast<arrow::Int64Array>(record_batch_->GetColumnByName(name));
+      if (!column) {
+        // TODO: Need to handle column that has NOT NULL attribute
```
Comment (kou) [inline ha_my_quiver.hpp:126]: We can't detect it by this branch.
This branch just detects the followings:
* A column named `name` doesn't exist
* Column type mismatch in MySQL and Apache Arrow

We can detect  `NULL`-able by `arrow::Field::nullable()`: https://arrow.apache.org/docs/cpp/api/datatype.html#_CPPv4NK5arrow5Field8nullableEv

File: mysql-test/my-quiver/t/select_all.test [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+--disable_warnings
+DROP TABLE IF EXISTS select_all;
+--enable_warnings
+--copy_file  suite/my-quiver/select_all.parquet $MYSQLD_DATADIR/test/select_all.parquet
+
+CREATE TABLE select_all (
+  c1 int,
+  c2 int,
+  c3 int,
+  n1 int
```
Comment (kou) [inline mysql-test/my-quiver/t/select_all.test:12]: It's better that we use meaningful name for readability. For example:

```suggestion
  int8 int,
  int16 int,
  int32 int,
  nonexistent int
```

File: test/generate_test_data.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+})
+
+table = pa.Table.from_pandas(df)
+pq.write_table(table, 'mysql-test/my-quiver/count_star.parquet', row_group_size=1)
+
+
+df = pd.DataFrame({
+    'c1': [1, 2, 3],
+    'c2': [10, 20, 30],
+    'c3': [100, 200, 300],
```
Comment (kou) [inline test/generate_test_data.py:16]: How about using characteristic values like the followings?
* Zero
* Negative number
* Border number (-129, -128, 127, 128, 32767, 32768 and so on)

File: mysql-test/my-quiver/t/count_star.test [hunk 4]
```diff
@@ -0,0 +1,17 @@
+--source ../include/my-quiver/have-my-quiver.inc
+--source ../include/my-quiver/have-my-quiver-helper.inc
+--disable_warnings
+DROP TABLE IF EXISTS count_star;
+--enable_warnings
+--copy_file  suite/my-quiver/count_star.parquet $MYSQLD_DATADIR/test/count_star.parquet
```
Comment (kou) [inline mysql-test/my-quiver/t/count_star.test]:
```suggestion
--copy_file suite/my-quiver/count_star.parquet $MYSQLD_DATADIR/test/count_star.parquet
```

File: mysql-test/my-quiver/t/select_all.test [hunk 5]
```diff
@@ -0,0 +1,25 @@
+--source ../include/my-quiver/have-my-quiver.inc
+--source ../include/my-quiver/have-my-quiver-helper.inc
+--disable_warnings
+DROP TABLE IF EXISTS select_all;
+--enable_warnings
+--copy_file  suite/my-quiver/select_all.parquet $MYSQLD_DATADIR/test/select_all.parquet
```
Comment (kou) [inline mysql-test/my-quiver/t/select_all.test]:
```suggestion
--copy_file suite/my-quiver/select_all.parquet $MYSQLD_DATADIR/test/select_all.parquet
```

File: .gitignore [hunk 6]
```diff
@@ -1 +1,2 @@
 build
+*.parquet
```
Comment (kou) [inline .gitignore]: We assume that our test `*.parquet`s exist in `mysql-test/suite/`, right?

```suggestion
/mysql-test/suite/*.parquet
```

File: ha_my_quiver.hpp (hunk 1 above)
Comment (kanga333) [reply ha_my_quiver.hpp:126]: I thought it was necessary to raise an error when a MySQL column has a NOT NULL constraint, and there is no column in Arrow. Handling type mismatch is also necessary, as you mentioned. I will think about it later.

File: mysql-test/my-quiver/t/select_all.test (hunk 2 above)
Comment (kanga333) [reply mysql-test/my-quiver/t/select_all.test:12]: I intended that this case tests that MyQuiver can select multiple columns, including not existing data. Therefore I named columns one, two, and three. However, as you mentioned, we should use more meaningful names and data. I will refactor this case after implementing a feature that supports various types to test each types can select border value.

File: ha_my_quiver.hpp (hunk 1 above)
Comment (kanga333) [reply ha_my_quiver.hpp:126]: However, I realized that MyQuiver should not support NOT NULL constraint because it doesn't update or insert data.

**Pull request merged**